couchbase-entity/src/main/java/com/kaufland/model/field/CblFieldHolder.kt

```diff
@@ -132,7 +132,11 @@ class CblFieldHolder(field: Field, allWrappers: List<String>) : CblBaseFieldHold
 
     private fun evaluateClazzForTypeConversion(): TypeName {
         return if (isIterable) {
-            TypeUtil.string()
+            if (TypeUtil.isMap(fieldType)) {
```

Comment:
this is the main change of this PR, isn't it? (the one relevant for supporting lists of ENUMs)